editor/src/components/canvas/dom-walker.ts

```diff
@@ -291,17 +292,25 @@ export function resubscribeObservers(domWalkerMutableState: {
   mutationObserver: MutationObserver
   resizeObserver: ResizeObserver
 }) {
-  const canvasRootContainer = document.getElementById(CanvasContainerID)
+  let rootElementId = CanvasContainerID
+  let rootElement = document.getElementById(rootElementId)
+
+  if (rootElement == null) {
+    // when there is a build error the canvas root container is not rendered, so we attach the observers to the canvas wrapper component
```

Comment:
hmm, why not use `canvas-container-outer` or another div which is wrapping the CanvasContainer?